operator/src/lib.rs

```diff
@@ -131,42 +131,44 @@ impl NifiState {
         data.insert("nifi.properties".to_string(), nifi_properties);
         data.insert("state-management.xml".to_string(), state_management_xml);
 
+        // And now create the actual ConfigMap
+        let config_map = stackable_operator::config_map::create_config_map(
+            &self.context.resource,
+            &cm_name,
+            data.clone(),
+        )?;
+
         match self
             .context
             .client
             .get::<ConfigMap>(cm_name, Some(&self.context.namespace()))
             .await
         {
-            Ok(config_map) => {
-                if let Some(existing_config_map_data) = config_map.data {
+            Ok(existing_config_map) => {
```

Comment:
I'm not sure which version I like better but you could also do away with the inner `if let`

```
            Ok(ConfigMap {
                data: Some(existing_config_map_data),
                ..
            }) => {
                if existing_config_map_data == data {
                    debug!(
                        "ConfigMap [{}] already exists with identical data, skipping creation!",
                        cm_name
                    );
                } else {
                    debug!(
                        "ConfigMap [{}] already exists, but differs, recreating it!",
                        cm_name
                    );
                    self.context.client.update(&config_map).await?;
                }
            }
            Err(e) => {
                // TODO: This is shit, but works for now. If there is an actual error in comes with
                //   K8S, it will most probably also occur further down and be properly handled
                debug!("Error getting ConfigMap [{}]: [{:?}]", cm_name, e);
                self.context.client.create(&config_map).await?;
            }
            _ => {}
```

You could even pull the `existing_config_map_data == data` check into a match guard....

But as i said: Feel free to ignore, I think it doesn't really make a difference.

Reply:
I like this better, used the match guard as well.